onnxruntime/core/platform/windows/env.cc

```diff
@@ -473,22 +473,29 @@ class WindowsEnv : public Env {
 
   virtual Status LoadDynamicLibrary(const std::string& library_filename, void** handle) const override {
     *handle = ::LoadLibraryExA(library_filename.c_str(), nullptr, LOAD_WITH_ALTERED_SEARCH_PATH);
-    if (!*handle)
-      return common::Status(common::ONNXRUNTIME, common::FAIL, "Failed to load library");
-    return common::Status::OK();
+    if (!*handle) {
+      const auto error_code = GetLastError();
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to load library, error code: ", error_code);
+    }
+    return Status::OK();
   }
 
-  virtual common::Status UnloadDynamicLibrary(void* handle) const override {
-    if (::FreeLibrary(reinterpret_cast<HMODULE>(handle)) == 0)
-      return common::Status(common::ONNXRUNTIME, common::FAIL, "Failed to unload library");
-    return common::Status::OK();
+  virtual Status UnloadDynamicLibrary(void* handle) const override {
+    if (::FreeLibrary(reinterpret_cast<HMODULE>(handle)) == 0) {
```

Comment:
>0 [](start = 60, length = 1)

NULL

Reply:
I didn't change that - but it does seem like 0 is correct -  "If the function fails, the return value is zero."  - https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-freelibrary